indico/modules/events/abstracts/controllers/management.py

```diff
@@ -279,3 +282,46 @@ def _process(self):
         sorted_abstracts = sorted(self.abstracts, key=attrgetter('friendly_id'))
         pdf = ConfManagerAbstractsToPDF(self.event_new, sorted_abstracts)
         return send_file('abstracts.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHManageReviewingRoles(RHManageAbstractsBase):
+    """Configure track roles (reviewers/conveners)."""
+
+    def _process(self):
+        roles = get_roles_for_event(self.event_new)
+        form = AbstractReviewingRolesForm(event=self.event_new, obj=FormDefaults(roles=roles))
+        conveners = set()
+        reviewers = set()
+
+        if form.validate_on_submit():
+            # Handle global reviewers/conveners
+            if '' in form.roles.data:
+                role_data = form.roles.data.pop('')
+                self.event_new.global_conveners = set(User.find(User.id.in_(role_data['convener'])))
+                self.event_new.abstract_global_reviewers = set(User.find(User.id.in_(role_data['reviewer'])))
+                conveners |= self.event_new.global_conveners
+                reviewers |= self.event_new.abstract_global_reviewers
+
+            track_dict = {track.id: track for track in Track.find(Track.id.in_(form.roles.data))}
+            user_dict = dict(get_users_in_roles(form.roles.data))
+            # Update track-specific reviewers/conveners
+            for track_id, roles in form.roles.data.viewitems():
+                track = track_dict[int(track_id)]
+                if track.event_new != self.event_new:
+                    raise Forbidden('Cannot manage track {} from event {}'.format(track, self.event_new))
```

Comment:
I have the feeling most of the logic in here actually belongs in the Field, not the Form or RH.

The field could return a tuple with the two reviewer lists and the necessary ACL role updates and then we'd just set/apply them here.